src/main/java/carpet/helpers/OptimizedExplosion.java

```diff
@@ -154,6 +161,11 @@ public static void doExplosionA(Explosion e, ExplosionLogHelper eLogger) {
                             densityCache.put(pair, density);
                         }
 
+                        // If it is needed, it saves the entity
+                        if (EXPLOSION.isNeeded()) {
```

Comment:
I would extract here boolean event = EXPLOSION.isNeeded();
and just use if (event)
```suggestion
                        if (eventNeeded) {
```

Reply:
Is there any benefit? 🤔